### PR TITLE
Feature/facilitates direct data saving for third-party modules

### PR DESCRIPTION
**Pull request by jungeunyoon**

### CODE REVIEW

**Comment by jungeunyoon** [comment]
[Implement common caching functionality [3]](https://trello.com/c/AfwNaXm8/4612-implement-common-caching-functionality-3)

File: data-saver.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+module.exports = (moduleName)=>{
+  return {
+    init() {
+      return platform.mkdir(getPathToCacheDir(moduleName));
+    },
+    putFile(componentId, data) {
+      return new Promise((resolve, reject)=>{
+        platform.writeTextFile(getPathToFile(moduleName, componentId), data)
+        .then(()=>{
```
Comment (santiagonoguez) [inline data-saver.js]: https://github.com/Rise-Vision/rise-common-electron/blob/master/platform.js#L239
writeTextFile() already returns a promise, you don't have to create a new one, just return it.

File: data-saver.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        });
+      });
+    },
+    getFile(componentId) {
+      return new Promise((resolve, reject)=>{
+        const pathToFile = getPathToFile(moduleName, componentId);
+
+        if (platform.fileExists(pathToFile)) {
+          platform.readTextFile(pathToFile)
+          .then((data)=>{
```
Comment (santiagonoguez) [inline data-saver.js]: same with reatTextFile() as before.
also platform.fileExists is synchronous, but I think is better to use a pure async approach if these functions will be used in concurrent scenarios.

File: data-saver.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        }
+      });
+    },
+    removeFile(componentId) {
+      return new Promise((resolve, reject)=>{
+        const pathToFile = getPathToFile(moduleName, componentId);
+
+        if (platform.fileExists(pathToFile)) {
+          platform.deleteRecursively(pathToFile)
+          .then(()=>{
```
Comment (santiagonoguez) [inline data-saver.js]: same as before

**Comment by jungeunyoon** [comment]
@Rise-Vision/delivery @Rise-Vision/creative Closing this PR until we investigate further into Chrome OS Player + Twitter Component Proof of Concept.

**Pull request closed without merging**